docs/gallery/general/read_basics.py

```diff
@@ -0,0 +1,374 @@
+`Analysis and Visualization Tools <https://nwb-overview.readthedocs.io/en/latest/tools/tools_home.html#analysis-and-visualization-tools>`_
+documentation for more details on the available tools.
+
+
+.. seealso::
+    You can learn more about the :py:class:`~pynwb.file.NWBFile` format in the :ref:`basics` tutorial.
+
+
+The following examples will reference variables that may not be defined within the block they are used in. For
+clarity, we define them here:
+"""
+import numpy as np
+from pynwb import NWBHDF5IO
+import matplotlib.pyplot as plt
+
+####################
+# Read the data
+# -------------
+# We will use the `DANDI <https://gui.dandiarchive.org/#/>`_ neurophysiology data archive
+# to access an NWB File. We will use data from one session of an experiment by
+# `Chandravadia et al. (2020) <https://www.nature.com/articles/s41597-020-0415-9>`_, where
+# the authors recorded single neuron activity from the medial temporal lobes of human subjects
+# while they performed a recognition memory task.
+#
+# Download the data
+# ^^^^^^^^^^^^^^^^^
+# First, we will demonstrate how to download an NWB data file from `DANDI <https://gui.dandiarchive.org/#/>`_
+# to your machine.
+#
+# 1. Go to the DANDI archive and open `this <https://gui.dandiarchive.org/#/dandiset/000004/draft>`_ dataset
+# 2. List the files in this dataset by clicking the "Files" button in Dandiset Actions (top right column within the page).
+#
+# .. image:: ../../_static/demo_dandi_view_files_in_dataset.png
+#   :width: 700
+#   :alt: view files on dandi
+#   :align: center
+#
+# 3. Choose the folder "sub-P11MHM" by clicking on its name.
+#
+# .. image:: ../../_static/demo_dandi_select_folder.png
+#   :width: 700
+#   :alt: selecting a folder on dandi
+#   :align: center
+#
+# 4. Download the NWB data file "sub-P11HMH_ses-20061101_ecephys+image.nwb" to your
+# computer by clicking on the download symbol.
+#
+# .. image:: ../../_static/demo_dandi_download_data.png
+#   :width: 700
+#   :alt: selecting a folder on dandi
+#   :align: center
+#
+# Stream the data
+# ^^^^^^^^^^^^^^^
+#
+# Next, we will demonstrate how to stream the data from the DANDI archive without
+# having to download it to your machine.
+# Streaming data requires having HDF5 installed with the ROS3 (read-only S3) driver.
+# You can install from `conda-forge <https://conda-forge.org>`_ using ``conda``.
+# You might need to first uninstall a currently installed version of ``h5py``.
+#
+# .. code-block:: bash
+#
+#    $ pip uninstall h5py
+#    $ conda install -c conda-forge "h5py>=3.2"
+#
+# We can access the data stored in an S3 bucket using the DANDI API,
+# which can be installed from pip:
+#
+# .. code-block:: bash
+#
+#    $ pip install -U dandi
+#
+# .. seealso::
+#     You can learn more about streaming data in the :ref:`streaming` tutorial.
+#
+# Then, we will use the ``DandiAPIClient`` to obtain the S3 URL that points to the NWB File
```

Comment:
can we use intersphinx to link to the [API documentation for DANDIAPIClient](https://dandi.readthedocs.io/en/stable/modref/dandiapi.html#dandi.dandiapi.DandiAPIClient)?

Reply:
oh sure, I'll add it!